csharp/src/Ice/InputStream.cs

```diff
@@ -165,7 +165,7 @@ public Encoding StartEncapsulation()
             Debug.Assert(_mainEncaps == null && _endpointEncaps == null);
             (Encoding Encoding, int Size) encapsHeader = ReadEncapsulationHeader();
             _mainEncaps = new Encaps(_limit, Encoding, encapsHeader.Size);
-            Debug.Assert(!encapsHeader.Encoding.Equals(Util.Encoding_1_0));
+            Debug.Assert(!encapsHeader.Encoding.Equals(Encoding.V1_0));
```

Comment:
Why do we have this assert here in the input stream? Shouldn't this assert instead be in the callers when un-marshal a request or response?

Reply:
Will fix in a future PR when I eliminate the 1.0 encoding entirely.

Reply:
I've just removed the remnants of 1.0 encoding support with this PR.

When we refactor InputStream into IncomingFrame + InputStream and OutputStream into OutgoingFrame + OutputStream, we should only check if the encoding is supported (meaning readable/writable) in the constructors of InputStream and OutputStream. If you're creating an InputStream resp. OutputStream, it's because you want to read resp. write with the specified encoding. If you're just transferring frame payloads that you don't need to read/write, then you don't need to create an InputStream or OutputStream for these payloads.